src/ucp/wireup/wireup.c

```diff
@@ -808,7 +799,38 @@ ucp_wireup_connect_lane_to_iface(ucp_ep_h ep, ucp_lane_index_t lane,
         return status;
     }
 
-    ucp_wireup_assign_lane(ep, lane, uct_ep, "");
+    if (ep->uct_eps[lane] == NULL) {
+        if (ucp_ep_has_cm_lane(ep)) {
+            /* create wireup EP in case of CM lane is used, since a WIREUP EP is
+             * used to keep user's pending requests and send WIREUP MSGs (if it
+             * is WIREUP MSG lane) until CM and WIREUP_MSG phases are done */
+            status = ucp_wireup_ep_create(ep, &ep->uct_eps[lane]);
+            if (status != UCS_OK) {
+                /* coverity[leaked_storage] */
+                return status;
+            }
+            ucp_wireup_ep_lane_set_next_ep(ep, lane, uct_ep);
+        } else {
+            ucs_trace("ep %p: assign uct_ep[%d]=%p", ep, lane, uct_ep);
+            ep->uct_eps[lane] = uct_ep;
+        }
+    } else {
+        /* If EP already exists, it's a wireup proxy, and we need to update
+         * its next_ep instead of replacing it.
+         */
+        ucs_assert(ucp_wireup_ep_test(ep->uct_eps[lane]));
+        ucs_assert(ucp_proxy_ep_extract(ep->uct_eps[lane]) == NULL);
+        ucs_assert(ucp_ep_has_cm_lane(ep) ||
```

Comment:
can this happen for new CM flow ?
e.g is it possible that ucp_ep_has_cm_lane() is true, and ep->uct_eps[lane]!=NULL?

Reply:
> can this happen for new CM flow ?
> e.g is it possible that ucp_ep_has_cm_lane() is true, and ep->uct_eps[lane]!=NULL?

`uct_eps[lane] != NULL` on a client side
`uct_eps[lane] == NULL` on a server side

Reply:
> `uct_eps[lane] != NULL` on a client side

Is it because of endpoint reconfiguration?

Reply:
no, this is the initial configuration for client and server.
during reconfiguration, it could be `!= NULL`, or `== NULL` depending on the case (re-used or new TL)